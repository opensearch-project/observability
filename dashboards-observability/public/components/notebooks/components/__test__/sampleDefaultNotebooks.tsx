/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

// Sample notebook with all input and output
export const sampleNotebook1 = {
  id: 'note_5f4b9eed-5898-4b39-ba6c-755c0fadd84e',
  dateCreated: '2020-08-20T18:00:59.845Z',
  name: 'test 1',
  dateModified: '2020-08-20T18:37:56.844Z',
  pluginVersion: '7.9.0',
  backend: 'Default',
  paragraphs: [
    {
      output: [{ result: '# Type your input here', outputType: 'MARKDOWN', execution_time: '0s' }],
      input: { inputText: '# Type your input here', inputType: 'MARKDOWN' },
      dateCreated: '2020-08-20T18:00:59.845Z',
      dateModified: '2020-08-20T18:00:59.845Z',
      id: 'paragraph_1a710988-ec19-4caa-83cc-38eb609427d1',
    },
    {
      output: [{ result: '# add code here\n', outputType: 'MARKDOWN', execution_time: '0s' }],
      input: { inputText: '# add code here\n', inputType: 'MARKDOWN' },
      dateCreated: '2020-08-20T18:01:07.662Z',
      dateModified: '2020-08-20T18:01:18.295Z',
      id: 'paragraph_c3107b15-da7d-4836-aef4-0996abbc8ab2',
    },
    {
      output: [{ result: '', outputType: 'VISUALIZATION', execution_time: '0s' }],
      input: {
        inputText:
          '{"viewMode":"view","panels":{"1":{"gridData":{"x":1,"y":0,"w":44,"h":20,"i":"1"},"type":"visualization","explicitInput":{"id":"1","savedObjectId":"935afa20-e0cd-11e7-9d07-1398ccfcefa3","vis":null}}},"isFullScreenMode":false,"filters":[],"useMargins":false,"id":"i3ccc6260-e314-11ea-9f99-b37e94bb02ca","timeRange":{"to":"2020-08-20T18:37:44.710Z","from":"2020-07-21T18:37:44.710Z"},"title":"embed_viz_i3ccc6260-e314-11ea-9f99-b37e94bb02ca","query":{"query":"","language":"lucene"},"refreshConfig":{"pause":true,"value":15}}',
        inputType: 'VISUALIZATION',
      },
      dateCreated: '2020-08-20T18:37:44.809Z',
      dateModified: '2020-08-20T18:37:56.844Z',
      id: 'paragraph_6d3237a9-6486-4f93-aa25-0a1c838faabd',
    },
  ],
};

// Parsed Output of sample notebook1
export const sampleParsedParagraghs1 = [
  {
    uniqueId: 'paragraph_1a710988-ec19-4caa-83cc-38eb609427d1',
    isRunning: false,
    inQueue: false,
    ishovered: false,
    isSelected: false,
    isInputHidden: false,
    isOutputHidden: false,
    showAddPara: false,
    isVizualisation: false,
    vizObjectInput: '',
    id: 1,
    inp: '# Type your input here',
    lang: 'text/x-md',
    editorLanguage: 'md',
    typeOut: ['MARKDOWN'],
    out: ['# Type your input here'],
  },
  {
    uniqueId: 'paragraph_c3107b15-da7d-4836-aef4-0996abbc8ab2',
    isRunning: false,
    inQueue: false,
    ishovered: false,
    isSelected: false,
    isInputHidden: false,
    isOutputHidden: false,
    showAddPara: false,
    isVizualisation: false,
    vizObjectInput: '',
    id: 2,
    inp: '# add code here\n',
    lang: 'text/x-md',
    editorLanguage: 'md',
    typeOut: ['MARKDOWN'],
    out: ['# add code here\n'],
  },
  {
    uniqueId: 'paragraph_6d3237a9-6486-4f93-aa25-0a1c838faabd',
    isRunning: false,
    inQueue: false,
    ishovered: false,
    isSelected: false,
    isInputHidden: false,
    isOutputHidden: false,
    showAddPara: false,
    isVizualisation: true,
    vizObjectInput:
      '{"viewMode":"view","panels":{"1":{"gridData":{"x":1,"y":0,"w":44,"h":20,"i":"1"},"type":"visualization","explicitInput":{"id":"1","savedObjectId":"935afa20-e0cd-11e7-9d07-1398ccfcefa3","vis":null}}},"isFullScreenMode":false,"filters":[],"useMargins":false,"id":"i3ccc6260-e314-11ea-9f99-b37e94bb02ca","timeRange":{"to":"2020-08-20T18:37:44.710Z","from":"2020-07-21T18:37:44.710Z"},"title":"embed_viz_i3ccc6260-e314-11ea-9f99-b37e94bb02ca","query":{"query":"","language":"lucene"},"refreshConfig":{"pause":true,"value":15}}',
    id: 3,
    inp:
      '{"viewMode":"view","panels":{"1":{"gridData":{"x":1,"y":0,"w":44,"h":20,"i":"1"},"type":"visualization","explicitInput":{"id":"1","savedObjectId":"935afa20-e0cd-11e7-9d07-1398ccfcefa3","vis":null}}},"isFullScreenMode":false,"filters":[],"useMargins":false,"id":"i3ccc6260-e314-11ea-9f99-b37e94bb02ca","timeRange":{"to":"2020-08-20T18:37:44.710Z","from":"2020-07-21T18:37:44.710Z"},"title":"embed_viz_i3ccc6260-e314-11ea-9f99-b37e94bb02ca","query":{"query":"","language":"lucene"},"refreshConfig":{"pause":true,"value":15}}',
    lang: 'text/x-',
    editorLanguage: '',
    typeOut: ['VISUALIZATION'],
    out: [''],
  },
];

// Sample notebook with all input and cleared outputs
export const sampleNotebook2 = {
  id: 'note_5f4b9eed-5898-4b39-ba6c-755c0fadd84e',
  dateCreated: '2020-08-20T18:00:59.845Z',
  name: 'test 1',
  dateModified: '2020-08-20T19:13:06.509Z',
  pluginVersion: '7.9.0',
  backend: 'Default',
  paragraphs: [
    {
      output: [
        {
          result:
            '# Type no output here\n* Sample link: [link](https://opensearch.org/)\n* ~~Strike~~, **Bold**, __Italic__',
          outputType: 'MARKDOWN',
          execution_time: '0s',
        },
      ],
      input: {
        inputText:
          '# Type no output here\n* Sample link: [link](https://opensearch.org/)\n* ~~Strike~~, **Bold**, __Italic__',
        inputType: 'MARKDOWN',
      },
      dateCreated: '2020-08-20T18:00:59.845Z',
      dateModified: '2020-08-20T19:13:06.509Z',
      id: 'paragraph_1a710988-ec19-4caa-83cc-38eb609427d1',
    },
    {
      output: [],
      input: { inputText: '# no output here\n', inputType: 'MARKDOWN' },
      dateCreated: '2020-08-20T18:01:07.662Z',
      dateModified: '2020-08-20T19:09:19.210Z',
      id: 'paragraph_c3107b15-da7d-4836-aef4-0996abbc8ab2',
    },
    {
      output: [],
      input: {
        inputText:
          '{"viewMode":"view","panels":{"1":{"gridData":{"x":1,"y":0,"w":44,"h":20,"i":"1"},"type":"visualization","explicitInput":{"id":"1","savedObjectId":"935afa20-e0cd-11e7-9d07-1398ccfcefa3","vis":null}}},"isFullScreenMode":false,"filters":[],"useMargins":false,"id":"i3ccc6260-e314-11ea-9f99-b37e94bb02ca","timeRange":{"to":"2020-08-20T18:37:44.710Z","from":"2020-07-21T18:37:44.710Z"},"title":"embed_viz_i3ccc6260-e314-11ea-9f99-b37e94bb02ca","query":{"query":"","language":"lucene"},"refreshConfig":{"pause":true,"value":15}}',
        inputType: 'VISUALIZATION',
      },
      dateCreated: '2020-08-20T18:37:44.809Z',
      dateModified: '2020-08-20T18:37:56.844Z',
      id: 'paragraph_6d3237a9-6486-4f93-aa25-0a1c838faabd',
    },
  ],
};

// Parsed Output of sample notebook2
export const sampleParsedParagraghs2 = [
  {
    uniqueId: 'paragraph_1a710988-ec19-4caa-83cc-38eb609427d1',
    isRunning: false,
    inQueue: false,
    ishovered: false,
    isSelected: false,
    isInputHidden: false,
    isOutputHidden: false,
    showAddPara: false,
    isVizualisation: false,
    vizObjectInput: '',
    id: 1,
    inp:
      '# Type no output here\n* Sample link: [link](https://opensearch.org/)\n* ~~Strike~~, **Bold**, __Italic__',
    lang: 'text/x-md',
    editorLanguage: 'md',
    typeOut: ['MARKDOWN'],
    out: [
      '# Type no output here\n* Sample link: [link](https://opensearch.org/)\n* ~~Strike~~, **Bold**, __Italic__',
    ],
  },
  {
    uniqueId: 'paragraph_c3107b15-da7d-4836-aef4-0996abbc8ab2',
    isRunning: false,
    inQueue: false,
    ishovered: false,
    isSelected: false,
    isInputHidden: false,
    isOutputHidden: false,
    showAddPara: false,
    isVizualisation: false,
    vizObjectInput: '',
    id: 2,
    inp: '# no output here\n',
    lang: 'text/x-md',
    editorLanguage: 'md',
    typeOut: [],
    out: [],
  },
  {
    uniqueId: 'paragraph_6d3237a9-6486-4f93-aa25-0a1c838faabd',
    isRunning: false,
    inQueue: false,
    ishovered: false,
    isSelected: false,
    isInputHidden: false,
    isOutputHidden: false,
    showAddPara: false,
    isVizualisation: true,
    vizObjectInput:
      '{"viewMode":"view","panels":{"1":{"gridData":{"x":1,"y":0,"w":44,"h":20,"i":"1"},"type":"visualization","explicitInput":{"id":"1","savedObjectId":"935afa20-e0cd-11e7-9d07-1398ccfcefa3","vis":null}}},"isFullScreenMode":false,"filters":[],"useMargins":false,"id":"i3ccc6260-e314-11ea-9f99-b37e94bb02ca","timeRange":{"to":"2020-08-20T18:37:44.710Z","from":"2020-07-21T18:37:44.710Z"},"title":"embed_viz_i3ccc6260-e314-11ea-9f99-b37e94bb02ca","query":{"query":"","language":"lucene"},"refreshConfig":{"pause":true,"value":15}}',
    id: 3,
    inp:
      '{"viewMode":"view","panels":{"1":{"gridData":{"x":1,"y":0,"w":44,"h":20,"i":"1"},"type":"visualization","explicitInput":{"id":"1","savedObjectId":"935afa20-e0cd-11e7-9d07-1398ccfcefa3","vis":null}}},"isFullScreenMode":false,"filters":[],"useMargins":false,"id":"i3ccc6260-e314-11ea-9f99-b37e94bb02ca","timeRange":{"to":"2020-08-20T18:37:44.710Z","from":"2020-07-21T18:37:44.710Z"},"title":"embed_viz_i3ccc6260-e314-11ea-9f99-b37e94bb02ca","query":{"query":"","language":"lucene"},"refreshConfig":{"pause":true,"value":15}}',
    lang: 'text/x-',
    editorLanguage: '',
    typeOut: [],
    out: [],
  },
];

// Sample notebook with no input
export const sampleNotebook3 = {
  id: 'note_5f4b9eed-5898-4b39-ba6c-755c0fadd84e',
  dateCreated: '2020-08-20T18:00:59.845Z',
  name: 'test 1',
  dateModified: '2020-08-20T19:13:06.509Z',
  pluginVersion: '7.9.0',
  backend: 'Default',
  paragraphs: [
    {
      output: [
        {
          result:
            '# Type no output here\n* Sample link: [link](https://opensearch.org/)\n* ~~Strike~~, **Bold**, __Italic__',
          outputType: 'MARKDOWN',
          execution_time: '0s',
        },
      ],
      dateCreated: '2020-08-20T18:00:59.845Z',
      dateModified: '2020-08-20T19:13:06.509Z',
      id: 'paragraph_1a710988-ec19-4caa-83cc-38eb609427d1',
    },
  ],
};

// Sample notebook with no paragraph id
export const sampleNotebook4 = {
  id: 'note_5f4b9eed-5898-4b39-ba6c-755c0fadd84e',
  dateCreated: '2020-08-20T18:00:59.845Z',
  name: 'test 1',
  dateModified: '2020-08-20T19:13:06.509Z',
  pluginVersion: '7.9.0',
  backend: 'Default',
  paragraphs: [
    {
      output: [
        {
          result:
            '# Type no output here\n* Sample link: [link](https://opensearch.org/)\n* ~~Strike~~, **Bold**, __Italic__',
          outputType: 'MARKDOWN',
          execution_time: '0s',
        },
      ],
      dateCreated: '2020-08-20T18:00:59.845Z',
      dateModified: '2020-08-20T19:13:06.509Z',
    },
  ],
};

// Sample notebook with no input and output
export const sampleNotebook5 = {
  id: 'note_5f4b9eed-5898-4b39-ba6c-755c0fadd84e',
  dateCreated: '2020-08-20T18:00:59.845Z',
  name: 'test 1',
  dateModified: '2020-08-20T19:13:06.509Z',
  pluginVersion: '7.9.0',
  backend: 'Default',
  paragraphs: [
    {
      dateCreated: '2020-08-20T18:00:59.845Z',
      dateModified: '2020-08-20T19:13:06.509Z',
      id: 'paragraph_c3107b15-da7d-4836-aef4-0996abbc8ab2',
    },
  ],
};
