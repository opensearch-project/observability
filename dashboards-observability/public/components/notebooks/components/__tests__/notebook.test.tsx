/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { fireEvent, render, waitFor } from '@testing-library/react';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { HttpResponse } from '../../../../../../../src/core/public';
import httpClientMock from '../../../../../test/__mocks__/httpClientMock';
import { sampleNotebook1 } from '../helpers/__tests__/sampleDefaultNotebooks';
import { Notebook } from '../notebook';

jest.mock('../../../../../../../src/plugins/embeddable/public', () => ({
  ViewMode: {
    EDIT: 'edit',
    VIEW: 'view',
  },
}));

// @ts-ignore
global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () =>
      Promise.resolve({
        status: {
          statuses: [{ id: 'plugin:reportsDashboards' }],
        },
      }),
  })
);

describe('<Notebook /> spec', () => {
  configure({ adapter: new Adapter() });

  it('renders the empty component', () => {
    const setBreadcrumbs = jest.fn();
    const renameNotebook = jest.fn();
    const cloneNotebook = jest.fn();
    const deleteNotebook = jest.fn();
    const setToast = jest.fn();
    const location = jest.fn();
    const history = jest.fn() as any;
    history.replace = jest.fn();
    const utils = render(
      <Notebook
        openedNoteId="mock-id"
        DashboardContainerByValueRenderer={jest.fn()}
        http={httpClientMock}
        parentBreadcrumb={{ href: 'parent-href', text: 'parent-text' }}
        setBreadcrumbs={setBreadcrumbs}
        renameNotebook={renameNotebook}
        cloneNotebook={cloneNotebook}
        deleteNotebook={deleteNotebook}
        setToast={setToast}
        location={location}
        history={history}
      />
    );
    expect(utils.container.firstChild).toMatchSnapshot();

    utils.getByText("Add code block").click();
    utils.getByText("Add visualization").click();
  });

  it('renders the component', async () => {
    const setBreadcrumbs = jest.fn();
    const renameNotebook = jest.fn();
    const cloneNotebook = jest.fn();
    const deleteNotebook = jest.fn();
    const setToast = jest.fn();
    const location = jest.fn();
    const history = jest.fn() as any;
    history.replace = jest.fn();
    httpClientMock.get = jest.fn(() =>
      Promise.resolve(({
        ...sampleNotebook1,
        path: sampleNotebook1.name,
        savedVisualizations: Array.from({ length: 5 }, (v, k) => ({
          label: `vis-${k}`,
          key: `vis-${k}`,
        })),
      } as unknown) as HttpResponse)
    );
    const utils = render(
      <Notebook
        openedNoteId={sampleNotebook1.id}
        DashboardContainerByValueRenderer={jest.fn()}
        http={httpClientMock}
        parentBreadcrumb={{ href: 'parent-href', text: 'parent-text' }}
        setBreadcrumbs={setBreadcrumbs}
        renameNotebook={renameNotebook}
        cloneNotebook={cloneNotebook}
        deleteNotebook={deleteNotebook}
        setToast={setToast}
        location={location}
        history={history}
      />
    );

    await waitFor(() => {
      expect(utils.container.firstChild).toMatchSnapshot();
    });
  });
});
