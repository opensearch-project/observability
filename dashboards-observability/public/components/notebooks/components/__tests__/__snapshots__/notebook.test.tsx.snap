// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notebook /> spec renders the component 1`] = `
<div
  style="float: left; width: 100%; max-width: 1500px;"
>
  <div
    class="euiPage euiPage--paddingMedium euiPage--grow"
  >
    <div
      class="euiPageBody euiPageBody--borderRadiusNone"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--justifyContentFlexEnd euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem"
        />
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        />
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        />
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <div
            class="euiPopover euiPopover--anchorDownCenter"
          >
            <div
              class="euiPopover__anchor"
            >
              <button
                class="euiButton euiButton--primary"
                data-test-subj="notebook-paragraph-actions-button"
                type="button"
              >
                <span
                  class="euiButtonContent euiButtonContent--iconRight euiButton__content"
                >
                  <svg
                    aria-hidden="true"
                    class="euiIcon euiIcon--medium euiIcon--inherit euiButtonContent__icon"
                    focusable="false"
                    height="16"
                    role="img"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M13.069 5.157L8.384 9.768a.546.546 0 01-.768 0L2.93 5.158a.552.552 0 00-.771 0 .53.53 0 000 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 000-.76.552.552 0 00-.771 0z"
                      fill-rule="non-zero"
                    />
                  </svg>
                  <span
                    class="euiButton__text"
                  >
                    Paragraph actions
                  </span>
                </span>
              </button>
            </div>
          </div>
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        />
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <div
            class="euiPopover euiPopover--anchorDownCenter"
          >
            <div
              class="euiPopover__anchor"
            >
              <button
                class="euiButton euiButton--primary"
                data-test-subj="notebook-notebook-actions-button"
                type="button"
              >
                <span
                  class="euiButtonContent euiButtonContent--iconRight euiButton__content"
                >
                  <svg
                    aria-hidden="true"
                    class="euiIcon euiIcon--medium euiIcon--inherit euiButtonContent__icon"
                    focusable="false"
                    height="16"
                    role="img"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M13.069 5.157L8.384 9.768a.546.546 0 01-.768 0L2.93 5.158a.552.552 0 00-.771 0 .53.53 0 000 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 000-.76.552.552 0 00-.771 0z"
                      fill-rule="non-zero"
                    />
                  </svg>
                  <span
                    class="euiButton__text"
                  >
                    Notebook actions
                  </span>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--s"
      />
      <h1
        class="euiTitle euiTitle--large"
      />
      <div
        class="euiSpacer euiSpacer--m"
      />
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsFlexStart euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <div
            class="euiText euiText--medium"
          >
            <div>
              <p>
                Created 
                <br />
                 
                Invalid date
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        style="float: left; width: 100%; max-width: 1130px; margin-top: 20px;"
      >
        <div
          class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
        >
          <div
            class="euiSpacer euiSpacer--xxl"
          />
          <div
            class="euiText euiText--medium"
          >
            <div
              class="euiTextAlign euiTextAlign--center"
            >
              <h2>
                No paragraphs
              </h2>
              <div
                class="euiText euiText--medium"
              >
                Add a paragraph to compose your document or story. Notebooks now support two types of input:
              </div>
            </div>
          </div>
          <div
            class="euiSpacer euiSpacer--xl"
          />
          <div
            class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--justifyContentSpaceEvenly euiFlexGroup--directionRow euiFlexGroup--responsive"
          >
            <div
              class="euiFlexItem euiFlexItem--flexGrow2"
            />
            <div
              class="euiFlexItem euiFlexItem--flexGrow3"
            >
              <div
                class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiCard euiCard--centerAligned euiCard--hasIcon"
              >
                <div
                  class="euiCard__top"
                >
                  <svg
                    aria-hidden="true"
                    class="euiIcon euiIcon--xxLarge euiCard__icon"
                    focusable="false"
                    height="16"
                    role="img"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M2.414 8.036L4.89 10.51a.5.5 0 01-.707.708L1.354 8.389a.5.5 0 010-.707l2.828-2.828a.5.5 0 11.707.707L2.414 8.036zm8.768 2.474l2.475-2.474-2.475-2.475a.5.5 0 01.707-.707l2.829 2.828a.5.5 0 010 .707l-2.829 2.829a.5.5 0 11-.707-.708zM8.559 2.506a.5.5 0 01.981.19L7.441 13.494a.5.5 0 01-.981-.19L8.559 2.506z"
                    />
                  </svg>
                </div>
                <div
                  class="euiCard__content"
                >
                  <span
                    class="euiTitle euiTitle--small euiCard__title"
                    id="random_html_idTitle"
                  >
                    Code block
                  </span>
                  <div
                    class="euiText euiText--small euiCard__description"
                    id="random_html_idDescription"
                  >
                    <p>
                      Write contents directly using markdown, SQL or PPL.
                    </p>
                  </div>
                </div>
                <div
                  class="euiCard__footer"
                >
                  <button
                    class="euiButton euiButton--primary"
                    style="margin-bottom: 17px;"
                    type="button"
                  >
                    <span
                      class="euiButtonContent euiButton__content"
                    >
                      <span
                        class="euiButton__text"
                      >
                        Add code block
                      </span>
                    </span>
                  </button>
                </div>
              </div>
            </div>
            <div
              class="euiFlexItem euiFlexItem--flexGrow3"
            >
              <div
                class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiCard euiCard--centerAligned euiCard--hasIcon"
              >
                <div
                  class="euiCard__top"
                >
                  <svg
                    aria-hidden="true"
                    class="euiIcon euiIcon--xxLarge euiCard__icon"
                    focusable="false"
                    height="16"
                    role="img"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M3 13h10V9.913l-2.571-2.826L8.56 8.753a.5.5 0 01-.728-.067L4.448 4.317 3 6.191V13zm5.295-5.35l1.837-1.64a.5.5 0 01.703.037l3.035 3.336a.5.5 0 01.13.337v3.78a.5.5 0 01-.5.5h-11a.5.5 0 01-.5-.5V6.02a.5.5 0 01.104-.305l1.947-2.52a.5.5 0 01.791-.001L8.295 7.65zM1 15h13.5a.5.5 0 110 1H.5a.5.5 0 01-.5-.5v-14a.5.5 0 011 0V15z"
                    />
                  </svg>
                </div>
                <div
                  class="euiCard__content"
                >
                  <span
                    class="euiTitle euiTitle--small euiCard__title"
                    id="random_html_idTitle"
                  >
                    Visualization
                  </span>
                  <div
                    class="euiText euiText--small euiCard__description"
                    id="random_html_idDescription"
                  >
                    <p>
                      Import OpenSearch Dashboards or Observability visualizations to the notes.
                    </p>
                  </div>
                </div>
                <div
                  class="euiCard__footer"
                >
                  <button
                    class="euiButton euiButton--primary"
                    style="margin-bottom: 17px;"
                    type="button"
                  >
                    <span
                      class="euiButtonContent euiButton__content"
                    >
                      <span
                        class="euiButton__text"
                      >
                        Add visualization
                      </span>
                    </span>
                  </button>
                </div>
              </div>
            </div>
            <div
              class="euiFlexItem euiFlexItem--flexGrow2"
            />
          </div>
          <div
            class="euiSpacer euiSpacer--xxl"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Notebook /> spec renders the empty component 1`] = `
<div
  style="float: left; width: 100%; max-width: 1500px;"
>
  <div
    class="euiPage euiPage--paddingMedium euiPage--grow"
  >
    <div
      class="euiPageBody euiPageBody--borderRadiusNone"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--justifyContentFlexEnd euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem"
        />
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        />
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        />
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <div
            class="euiPopover euiPopover--anchorDownCenter"
          >
            <div
              class="euiPopover__anchor"
            >
              <button
                class="euiButton euiButton--primary"
                data-test-subj="notebook-paragraph-actions-button"
                type="button"
              >
                <span
                  class="euiButtonContent euiButtonContent--iconRight euiButton__content"
                >
                  <svg
                    aria-hidden="true"
                    class="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                    focusable="false"
                    height="16"
                    role="img"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  />
                  <span
                    class="euiButton__text"
                  >
                    Paragraph actions
                  </span>
                </span>
              </button>
            </div>
          </div>
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        />
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <div
            class="euiPopover euiPopover--anchorDownCenter"
          >
            <div
              class="euiPopover__anchor"
            >
              <button
                class="euiButton euiButton--primary"
                data-test-subj="notebook-notebook-actions-button"
                type="button"
              >
                <span
                  class="euiButtonContent euiButtonContent--iconRight euiButton__content"
                >
                  <svg
                    aria-hidden="true"
                    class="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiButtonContent__icon"
                    focusable="false"
                    height="16"
                    role="img"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  />
                  <span
                    class="euiButton__text"
                  >
                    Notebook actions
                  </span>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--s"
      />
      <h1
        class="euiTitle euiTitle--large"
      />
      <div
        class="euiSpacer euiSpacer--m"
      />
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsFlexStart euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <div
            class="euiText euiText--medium"
          >
            <div>
              <p>
                Created 
                <br />
                 
                Invalid date
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        style="float: left; width: 100%; max-width: 1130px; margin-top: 20px;"
      >
        <div
          class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
        >
          <div
            class="euiSpacer euiSpacer--xxl"
          />
          <div
            class="euiText euiText--medium"
          >
            <div
              class="euiTextAlign euiTextAlign--center"
            >
              <h2>
                No paragraphs
              </h2>
              <div
                class="euiText euiText--medium"
              >
                Add a paragraph to compose your document or story. Notebooks now support two types of input:
              </div>
            </div>
          </div>
          <div
            class="euiSpacer euiSpacer--xl"
          />
          <div
            class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--justifyContentSpaceEvenly euiFlexGroup--directionRow euiFlexGroup--responsive"
          >
            <div
              class="euiFlexItem euiFlexItem--flexGrow2"
            />
            <div
              class="euiFlexItem euiFlexItem--flexGrow3"
            >
              <div
                class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiCard euiCard--centerAligned euiCard--hasIcon"
              >
                <div
                  class="euiCard__top"
                >
                  <svg
                    aria-hidden="true"
                    class="euiIcon euiIcon--xxLarge euiIcon-isLoading euiCard__icon"
                    focusable="false"
                    height="16"
                    role="img"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  />
                </div>
                <div
                  class="euiCard__content"
                >
                  <span
                    class="euiTitle euiTitle--small euiCard__title"
                    id="random_html_idTitle"
                  >
                    Code block
                  </span>
                  <div
                    class="euiText euiText--small euiCard__description"
                    id="random_html_idDescription"
                  >
                    <p>
                      Write contents directly using markdown, SQL or PPL.
                    </p>
                  </div>
                </div>
                <div
                  class="euiCard__footer"
                >
                  <button
                    class="euiButton euiButton--primary"
                    style="margin-bottom: 17px;"
                    type="button"
                  >
                    <span
                      class="euiButtonContent euiButton__content"
                    >
                      <span
                        class="euiButton__text"
                      >
                        Add code block
                      </span>
                    </span>
                  </button>
                </div>
              </div>
            </div>
            <div
              class="euiFlexItem euiFlexItem--flexGrow3"
            >
              <div
                class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiCard euiCard--centerAligned euiCard--hasIcon"
              >
                <div
                  class="euiCard__top"
                >
                  <svg
                    aria-hidden="true"
                    class="euiIcon euiIcon--xxLarge euiIcon-isLoading euiCard__icon"
                    focusable="false"
                    height="16"
                    role="img"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  />
                </div>
                <div
                  class="euiCard__content"
                >
                  <span
                    class="euiTitle euiTitle--small euiCard__title"
                    id="random_html_idTitle"
                  >
                    Visualization
                  </span>
                  <div
                    class="euiText euiText--small euiCard__description"
                    id="random_html_idDescription"
                  >
                    <p>
                      Import OpenSearch Dashboards or Observability visualizations to the notes.
                    </p>
                  </div>
                </div>
                <div
                  class="euiCard__footer"
                >
                  <button
                    class="euiButton euiButton--primary"
                    style="margin-bottom: 17px;"
                    type="button"
                  >
                    <span
                      class="euiButtonContent euiButton__content"
                    >
                      <span
                        class="euiButton__text"
                      >
                        Add visualization
                      </span>
                    </span>
                  </button>
                </div>
              </div>
            </div>
            <div
              class="euiFlexItem euiFlexItem--flexGrow2"
            />
          </div>
          <div
            class="euiSpacer euiSpacer--xxl"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
